compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1775,6 +1775,25 @@ impl<'a> Parser<'a> {
             self.sess.expr_parentheses_needed(&mut err, *sp, None);
         }
         err.span_label(span, "expected expression");
+        if self.prev_token.kind == TokenKind::BinOp(token::Plus)
+            && self.token.kind == TokenKind::BinOp(token::Plus)
+            && self.look_ahead(1, |t| !t.is_lit())
+        {
+            let span = self.prev_token.span.to(self.token.span);
+            err.note("Rust has no dedicated increment and decrement operators");
+            err.span_suggestion_verbose(
+                span,
+                "try using `+= 1` instead",
+                " += 1".into(),
+                Applicability::Unspecified,
+            );
+        } else if self.token.kind == TokenKind::BinOp(token::Plus)
+            && self.look_ahead(1, |t| t.kind == TokenKind::BinOp(token::Plus))
+            && self.look_ahead(2, |t| !t.is_lit())
+        {
+            err.note("Rust has no dedicated increment and decrement operators");
+            err.help("try using `+= 1` instead");
```

Comment:
```suggestion
            let lo = self.token.span; // `+`
            self.bump(); // `+`
            let pre_sp = lo.to(self.token.span);
            err.multipart_suggestion(
                "try using `+= 1` instead"
                vec![
                    (pre_sp, String::new()),
                    (self.token.span.shrink_to_hi(), " += 1".to_string()),
                ],
                Applicability::MachineApplicable,
            );
            self.bump(); // ident
```

Reply:
Won't this turn `++i` into `+= 1i`?

Reply:
Yes, good catch. You'll need to account for `++i` and `i++` specifically.